tests/std/tests/P2321R2_views_zip_transform/test.cpp

```diff
@@ -0,0 +1,864 @@
+                assert(static_cast<decltype(ranges::size(transformed_elements))>(
+                           ranges::distance(zipped_transformed_range.begin(), zipped_transformed_range.end()))
+                       == ranges::size(transformed_elements));
+
+                STATIC_ASSERT(noexcept(zipped_transformed_range.begin()) == noexcept(declval<InnerView&>().begin())
+                              && is_nothrow_move_constructible_v<ranges::iterator_t<InnerView>>);
+                STATIC_ASSERT(noexcept(zipped_transformed_range.end()) == noexcept(declval<InnerView&>().end())
+                              && is_nothrow_move_constructible_v<ranges::sentinel_t<InnerView>>);
+            }
+
+            STATIC_ASSERT(CanMemberBegin<const ZipTransformType> == has_const_begin_end);
+            STATIC_ASSERT(CanMemberEnd<const ZipTransformType> == has_const_begin_end);
+            if constexpr (has_const_begin_end) {
+                const same_as<ranges::iterator_t<const ZipTransformType>> auto begin_itr =
+                    as_const(zipped_transformed_range).begin();
+                [[maybe_unused]] const same_as<ranges::sentinel_t<const ZipTransformType>> auto end_sentinel =
+                    as_const(zipped_transformed_range).end();
+
+                assert(*begin_itr == *ranges::begin(transformed_elements));
+                assert(static_cast<decltype(ranges::size(transformed_elements))>(
+                           ranges::distance(zipped_transformed_range.begin(), zipped_transformed_range.end()))
+                       == ranges::size(transformed_elements));
+                STATIC_ASSERT(noexcept(as_const(zipped_transformed_range).begin())
+                                  == noexcept(declval<const InnerView&>().begin())
+                              && is_nothrow_move_constructible_v<ranges::iterator_t<const InnerView>>);
+                STATIC_ASSERT(
+                    noexcept(as_const(zipped_transformed_range).end()) == noexcept(declval<const InnerView&>().end())
+                    && is_nothrow_move_constructible_v<ranges::sentinel_t<const InnerView>>);
+            }
+
+            validate_iterators_sentinels<TransformType, TransformedElementsContainer,
+                decltype(zipped_transformed_range), RangeTypes...>(zipped_transformed_range, transformed_elements);
+            validate_iterators_sentinels<TransformType, TransformedElementsContainer,
+                decltype(as_const(zipped_transformed_range)), RangeTypes...>(
+                as_const(zipped_transformed_range), transformed_elements);
+        }
+    }
+
+    return true;
+}
+
+#pragma warning(pop)
+
+constexpr auto one_element_transform_closure = []<class ElementType>(const ElementType& a) { return (a * 5); };
+
+constexpr auto three_element_transform_closure = []<class Type1, class Type2, class Type3>(const Type1& a,
+                                                     const Type2& b, const Type3& c) { return (a * b + c); };

```

Comment:
Suggestion: As these template parameters aren't separately inspected, these generic lambdas could just take `const auto&`. (This creates one template parameter per function parameter.)